pkg/buildkit/worker/pool.go

```diff
@@ -441,6 +425,38 @@ func (p *workerPool) updateWorkers(ctx context.Context) error {
 	return err
 }
 
+// attempts to lease a pod, build and endpoint url, and provide a request result
+func (p *workerPool) processPodRequest(ctx context.Context, req *PodRequest, pod *corev1.Pod) (success bool) {
+	log := p.log.WithValues("podName", pod.Name)
+
+	log.Info("Attempting to lease pod")
+	if err := p.leasePod(ctx, pod, req.owner); err != nil {
+		log.Error(err, "Failed to lease pod")
+
+		req.result <- PodRequestResult{err: err}
+		return
+	}
+
+	log.Info("Building endpoint URL")
+	addr, err := p.buildEndpointURL(ctx, pod.Name)
+
+	if err != nil {
```

Comment:
How common would this type of race condition (pod running, but no endpoint) happen? I'd hate to fail the build for something relatively common

Reply:
This is not common but it did happen to Carlin a couple of times back when there was NOT a retry strategy in place. If we `GET` the record a few milliseconds before k8s can update it with the new, ready pod's address, then this would fail. The retry should handle this w/o issue but this is just a _defensive_ guard to ensure we don't leave pods leased indefinitely.

Make sense?

Reply:
🤦  forgot about the retry. Makes sense!